Source/Extensions/NSURLRequest.swift

```diff
@@ -0,0 +1,20 @@
+import Foundation
+
+public extension NSURLRequest {
+  var jsonPayload: NSDictionary {
+    let json = HTTPBody.flatMap { try? NSJSONSerialization.JSONObjectWithData($0, options: NSJSONReadingOptions(rawValue: 0)) }
+    return (json as? [String: AnyObject]) ?? [:]
+  }
+}
+
+public extension NSMutableURLRequest {
+  override var jsonPayload: NSDictionary {
+    get {
+      return super.jsonPayload
+    }
+
+    set {
+      HTTPBody = try? NSJSONSerialization.dataWithJSONObject(newValue, options: NSJSONWritingOptions(rawValue: 0))
```

Comment:
I think the issue I described in https://github.com/thoughtbot/Swish/pull/34#discussion_r42904628 applies here if the dictionary is empty, as shown in https://github.com/the-grid/Disc/commit/02b31d935192307749767c8b095aa11f0270e39a.

Maybe we need a `guard` for `newValue.isEmpty`?

cc @gfontenot @sidraval